### PR TITLE
Bugfix/212 invest log regex

### LINKED ISSUE #212: bug in regex for invest logfile markup
We markup invest logfiles based on which module is doing the logging, and our pattern-matching includes a regex like`new RegExp('natcap.invest.carbon')`. We need to be careful to only find matches for strings with those literal dots.

Something like `new RegExp('natcap.invest.carbon'.replace(/\./g, '\\\\.'))` should do it.

### PR DESCRIPTION
**Pull request by davemfish**
This PR fixes #212 .

It also generally simplifies and cleans up this feature by moving it to a separate module. Also, when this feature was first developed, we were parsing invest logs line-by-line whereas now we parse them message-by-message, and that makes it much easier (simpler regex) to assign the correct markup to multi-line messages such as Tracebacks.

Now the markup is as follows:
* `DEBUG|INFO` messages from the invest model are bold
* `WARNING` messages from the invest model are red (not bold)
* `ERROR|CRITICAL` messages from any module are red & bold

**BEFORE:**
logging_before1
the shapely warnings are mistakenly bold because they happen to include the string `natcap/invest/coastal_vulnerability`

logging_before2
And this one is incorrectly bold red because it includes the string `AttributeError`.

**AFTER:**
logging_after
shapely messages are plain. `WARNINGS` from the invest model are elevated to red.

### CODE REVIEW

File: src/renderer/components/InvestTab/index.jsx [hunk 1]
```diff
@@ -304,7 +304,6 @@ export default class InvestTab extends React.Component {
                   logfile={logfile}
                   executeClicked={executeClicked}
                   jobID={jobID}
-                  pyModuleName={modelSpec.pyname}
```
Comment (davemfish) [inline src/renderer/components/InvestTab/index.jsx:307]: `LogTab` component was not using this prop.

File: src/main/investLogMarkup.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  // eslint-disable-next-line
+  for (const [cls, pattern] of Object.entries(patterns)) {
+    if (pattern.test(message)) {
+      const markup = `<${logTextTag} class="${cls}">${message}</${logTextTag}>`;
+      return sanitizeHtml(markup, allowedHtml);
+    }
+  }
+  return sanitizeHtml(message, allowedHtml);
```
Comment (dcdenu4) [inline src/main/investLogMarkup.js:31]: I don't think it really matters, but if no markup has been done, do we need to pass `allowedHtml` in this instance?

**Review by dcdenu4 — APPROVED** [review]
Hey @davemfish , this looks good to me. I like breaking that bit of logic out into its own module. I had one question / comment but it's pretty trivial so I'm approving and will let you go ahead and merge if no changes need to be made.

File: src/main/investLogMarkup.js (hunk 2 above)
Comment (davemfish) [reply src/main/investLogMarkup.js]: Good one! Change pushed.

**Comment by davemfish** [comment]
> Hey @davemfish , this looks good to me. I like breaking that bit of logic out into its own module. I had one question / comment but it's pretty trivial so I'm approving and will let you go ahead and merge if no changes need to be made.

Thanks @dcdenu4 I made the change you suggested. Might as well let the tests run again.

**Review by dcdenu4 — APPROVED** [review]
Thanks @davemfish !

**Pull request merged**